backends/dpdk/dpdkHelpers.cpp

```diff
@@ -619,11 +628,20 @@ bool ConvertStatementToDpdk::preorder(const IR::MethodCallStatement *s) {
                 ::error("%1%: must be one of the existing errors", s);
             auto error_id = structure->error_map.at(error->expression->to<IR::Member>()->member);
             auto end_label = Util::printf_format("label_%dend", next_label_id++);
-            add_instr(new IR::DpdkJmpEqualStatement(
-                        end_label,
-                        condition->expression, new IR::BoolLiteral(false)));
-            add_instr(new IR::DpdkMovStatement(
-                        new IR::PathExpression("metadata"), new IR::Constant(error_id)));
+            if (condition->expression->is<IR::BoolLiteral>()) {
```

Comment:
are these two branches of the if swapped?

Reply:
No. This code is for the case where both the  operands of conditional jump are constant. It moves the first operand to a new temporary variable.
Actually, I think I can remove the conditional branch altogether for this case.

Reply:
@mbudiu-vmw Thanks for the review.. 
Could you please have a look at the updated code and see if it looks fine now?